hr_timesheet_holiday/models/hr_holidays.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Therp BV <http://therp.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import models, fields, api
+from openerp.exceptions import Warning as UserError
+
+
+class HrHolidaysStatus(models.Model):
+    """Add analytic account to holiday status"""
+    _inherit = 'hr.holidays.status'
+
+    analytic_account_id = fields.Many2one('account.analytic.account',
+        'Analytic Account')
+
+
+class HrHolidays(models.Model):
+    """Update analytic account on status change of Leave Request"""
+    _inherit = 'hr.holidays'
+
+    @api.multi
+    def holidays_validate(self):
+        res = super(HrHolidays, self).holidays_validate()
+
+        # Look for analytic account linked to Leave Type
+        for leave in self:
+            account = leave.holiday_status_id.analytic_account_id
+            if not account:
+                raise UserError("No analytic account defined for Leave Type '%s'" %
+                    (leave.holiday_status_id.name,))
+
+            # Look for company and hours per working day
+            if not leave.department_id:
+                raise UserError("Department not set")
+            company = leave.department_id.company_id
+            if not company:
```

Comment:
`hr.employee` inherits this field from `resource.resource` which sets an unconditional default: https://github.com/OCA/OCB/blob/8.0/addons/resource/resource.py#L679, so one at least has to try hard to unset it. Impossible via the standard UI anyways, so I think it's safe to assume this is set.